matRad_calcParticleDoseBixel.m

```diff
@@ -34,23 +36,45 @@
 %
 % %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
+% simplification since SSD is not exactly in y direction but sourceRashiDistance is
+% upstream value is used, since we then don't rashi need geometrical thickness
+rashiSurfaceDist = SSD - rangeShifter.sourceRashiDistance;
+
 % range shift
 depths = baseData.depths + baseData.offset;
 
+% adjust radDepth according to range shifter
+radDepths = radDepths + rangeShifter.eqThickness;
+
 % convert from MeV cm^2/g per primary to Gy mm^2 per 1e6 primaries
 conversionFactor = 1.6021766208e-02;
 
  % calculate initial focus sigma
 SigmaIni = matRad_interp1(baseData.initFocus.dist(focusIx,:)',baseData.initFocus.sigma(focusIx,:)',SSD);
 
+
+if exist('rangeShifter', 'var') && exist('radiationMode', 'var') &&  rangeShifter.eqThickness ~= 0
+  if isequal(radiationMode, 'protons')
+    sigmaRashi = matRad_sigmaRashi(baseData, rangeShifter.eqThickness, rashiSurfaceDist);
+  else % sigmaRashi = 0 for carbons
+    sigmaRashi = 0;
+  end
+  
+  else
+    sigmaRashi = 0;
+end
+
 if ~isfield(baseData,'sigma')
     
     % interpolate depth dose, sigmas, and weights    
     X = matRad_interp1(depths,[conversionFactor*baseData.Z baseData.sigma1 baseData.weight baseData.sigma2],radDepths);
     
+    % set dose for query > tabulated depth dose values to zero
+    X(radDepths >= max(depths),1) = 0;
```

Comment:
is this necessary if we need to extrapolate in line #70? we could also handle this with an exolicit extrapolation value passed to interp1.